core/src/test/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbeTest.java

```diff
@@ -150,4 +150,30 @@ void shouldRecognizeInvalidGitHubUrl() throws Exception {
         assertThat(r1.status()).isEqualTo(ResultStatus.FAILURE);
         assertThat(r1.message()).isEqualTo("The plugin SCM link is invalid");
     }
+
+    @Test
+    void shouldReturnCorrectScmFolderPath() throws IOException {
+        final Plugin p1 = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final GitHub github = mock(GitHub.class);
+        final String repositoryName = "jenkinsci/test-repo";
+
+        when(p1.getScm()).thenReturn("https://github.com/" + repositoryName);
+        when(p1.getDetails()).thenReturn(Map.of(
+            UpdateCenterPluginPublicationProbe.KEY, ProbeResult.success(UpdateCenterPluginPublicationProbe.KEY, "")
+        ));
+        when(ctx.getGitHub()).thenReturn(github);
+        GHRepository repository = ctx.getGitHub().getRepository(repositoryName);
+        when(github.getRepository(repositoryName)).thenReturn(repository);
```

Comment:
you are initializing an object with a call to a method that you mock just after the initializing. 
This doesn't seem right. 

If you need to get a `GHRepository` instance, then mock it. 

```suggestion
        GHRepository repository = mock(GHRepository.class);
        when(github.getRepository(repositoryName)).thenReturn(repository);
```

But first, why do you need that instance?

Reply:
To mock `            context.getGitHub().getRepository(matcher.group("repo"));` in `fromSCMLink` method